lib/screen_controller.dart

```diff
@@ -185,6 +188,51 @@ class ScreenController {
           },
         );
       }
+    } else if (action is PlaidLinkAction) {
+      if (scopeManager != null) {}
+      final linkToken = action.linkToken(dataContext);
+      if (linkToken != null) {
+        PlaidLinkController().openPlaidLink(
+          context,
+          linkToken,
+          (linkSuccess) {
+            if (action.onSuccess != null) {
+              executeActionWithScope(
+                context,
+                scopeManager!,
+                action.onSuccess!,
+                event: EnsembleEvent(action.initiator!,
+                    data: {'successData': linkSuccess.toJson()}),
+              );
+            }
+          },
+          (linkEvent) {
+            if (action.onEvent != null) {
+              executeActionWithScope(
+                context,
+                scopeManager!,
+                action.onEvent!,
+                event: EnsembleEvent(action.initiator!,
+                    data: {'eventData': linkEvent.toJson()}),
```

Comment:
by convention we always do `event.data.*`. So do `data: linkEvent.toJson()`. Same with others

Reply:
We have three callbacks with data, so I thought to split it evenly with their name like eventData, successData, errorData. Okay I'll add every object directly to the data param.

Reply:
Done